inferentia/scripts/gen_triton_model.py

```diff
@@ -336,7 +351,8 @@ def _validate_output_dict(self, expected_count):
     return init_impl
 
 
-def get_tensorflow_execute_impl():
+def get_tensorflow_execute_impl(enable_dynamic_batching,
```

Comment:
These python models doesnot do anything to enable triton's batching. When dyanmic batching in triton is enabled then Triton Server will itself provide multiple requests in one execute call if it was able to batch multiple requests. It is inefficient to run the extra logic even if the server was unable to get multiple requests for the python backend. 

enable_dynamic_batching should not appear in python models.. Instead it should rely on the requests list received. If the list contains single request then nothing extra needs to be done.

Reply:
When `enable_dynamic_batching` is specified though, Triton will send inputs of `[X, input dim]` according to my tests. eg: Input provided from perf_analyzer is [6, 4], model accepts [X, 4] but all the requests sent are of type [1,6,4]. 
This is why `squeeze` is required for batched requests.

However, if batching is not enabled, the inputs are [6,4].

Are you suggesting I squeeze the inputs regardless?

Reply:
Ohh.. I didn't see that before..  You should not design the python model on the basis of what perf analyzer generates. That is just a tool for running inferencing on server. Perf analyzer can be modified to generate requests expected by server but we must not loose generality.

So this is what your python model should be doing.. If the shape expected is in the following format: [batch size, X]. 
If the python model in its exec function gets three requests in the list as follows: [6, 4], [6,4] and [6, 4],
It should simply form a tensor of size [18, 4] and run that in neuron SDK. 

Now to test this you should run perf_analyzer as follows:

`perf_analyzer -m <model_name> --concurrency-range 10 -b 6` 

Note the model config should specify the dim as [4] and have max_batch_size specified as greater than or equal to 18. To form a batch of size 6, perf_analyzer goes through the tensors in a round-robin fashion.  Look at these documentation for more context.. https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#inputs-and-outputs
https://github.com/triton-inference-server/server/blob/main/docs/perf_analyzer.md#real-input-data